libminifi/src/core/ProcessSession.cpp

```diff
@@ -939,9 +939,7 @@ void ProcessSession::persistFlowFilesBeforeTransfer(
         if (ff->isStored() && flowFileRepo->Delete(ff->getUUIDStr())) {
           // original must be non-null since this flowFile is already stored in the repos ->
           // must have come from a session->get()
-          auto claim = original->getResourceClaim();
-          // decrement on behalf of the persisted-instance-to-be-deleted
-          if (claim) claim->decreaseFlowFileRecordOwnedCount();
```

Comment:
Could you explain this change? Why is it no longer necessary to decrease the refcount here?

Reply:
since `flowFileRepo->Delete` is async, the deletion from the content repository has to be the responsibility of the FlowFileRepository (as was before only with `removeIfOrphaned`), but as we would like to handle the removal of content automatically based on the refCount, that method has been removed, so now a `decreaseFlowFileRecordOwnedCount` will trigger a remove if the refCount goes to zero